File: cryze_android_app/app/src/main/java/com/tencentcs/iotvideo/messagemgr/ModelMessage.java
package com.tencentcs.iotvideo.messagemgr;

import androidx.annotation.NonNull;

/* loaded from: classes2.dex */
public class ModelMessage extends Message {
    public String payload;
    public String device;
    public String messageType;

    public ModelMessage(String str, long j10, int i10, int i11, String str2, String str3) {
        super(i10, j10, i11);
        this.device = str;
        this.messageType = str2;
        this.payload = str3;
    }

    @NonNull
    public String toString() {
        return "ModelMessage{device='" + this.device +
                "', path='" +
                this.messageType +
                "', error='" +
                this.error +
                "', data='" +
                this.payload +
                "'}";
    }
}

File: cryze_android_app/android/app/src/main/java/com/tencentcs/iotvideo/messagemgr/ModelMessage.kt
package com.tencentcs.iotvideo.messagemgr

import com.google.gson.Gson
import com.google.gson.GsonBuilder
import com.google.gson.JsonParser

open class ModelMessage(
    var device: String,
    id: Long,
    type: MessageType,
    error: Int,
    var path: String,
    var data: String
) : Message(type, id, error) {

    constructor(device: String, id: Long, type: Int, error: Int, path: String, data: String)
            : this(device, id, MessageType.fromInt(type), error, path, data)

    fun getPrettyMessage(): String {
        val gson: Gson = GsonBuilder().setPrettyPrinting().create()
        val jsonElement = JsonParser.parseString(this.data).asJsonObject

        // Add additional values to the JSON object
        jsonElement.addProperty("device", this.device)
        jsonElement.addProperty("path", this.path)
        jsonElement.addProperty("error", this.error)
        jsonElement.addProperty("id", this.id)
        jsonElement.addProperty("type", this.type.name)

        return gson.toJson(jsonElement)
    }

    override fun toString(): String {
        return "ModelMessage{device='" + this.device +
                "', path='" +
                this.path +
                "', error='" +
                this.error +
                "', data='" +
                this.data +
                "' " + super.toString() + "}"
    }
}
